pkg/controller/webhooks/codey.go

```diff
@@ -0,0 +1,131 @@
+
+//+kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list
+//+kubebuilder:rbac:groups=codey.io,resources=codeyinstances,verbs=get;list;watch;patch;update
+//+kubebuilder:rbac:groups=codey.io,resources=codeyinstances/status,verbs=get;list;watch;patch;update
+
+const (
+	codeyUrlSuffix = ".app.codey.ch"
+)
+
+var (
+	codeyGK = schema.GroupKind{Group: "codey.io", Kind: "CodeyInstance"}
+	codeyGR = schema.GroupResource{Group: codeyGK.Group, Resource: "codeyinstance"}
+)
+
+var _ webhook.CustomValidator = &CodeyInstanceWebhookHandler{}
+
+type CodeyInstanceWebhookHandler struct {
+	DefaultWebhookHandler
+}
+
+// SetupCodeyInstanceWebhookHandlerWithManager registers the validation webhook with the manager.
+func SetupCodeyInstanceWebhookHandlerWithManager(mgr ctrl.Manager, withQuota bool) error {
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&codey.CodeyInstance{}).
+		WithValidator(&CodeyInstanceWebhookHandler{
+			DefaultWebhookHandler: *New(
+				mgr.GetClient(),
+				mgr.GetLogger().WithName("webhook").WithName("codey"),
+				withQuota,
+				&codey.CodeyInstance{},
+				"codey",
+				codeyGK,
+				codeyGR,
+			),
+		}).
+		Complete()
+}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (n *CodeyInstanceWebhookHandler) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	warning, err := n.DefaultWebhookHandler.ValidateCreate(ctx, obj)
+	if warning != nil || err != nil {
+		return warning, err
+	}
+
+	codey, ok := obj.(*codey.CodeyInstance)
+	if !ok {
+		return nil, fmt.Errorf("provided manifest is not a valid CodeyInstance object")
+	}
+
+	codeyFqdn := codey.ObjectMeta.Name + codeyUrlSuffix
+	if err := isCodeyFqdnUnique(codeyFqdn); err != nil {
+		return nil, fmt.Errorf("failed FQDN validation: %v", err)
+	}
+
+	return nil, nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
+func (p *CodeyInstanceWebhookHandler) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	_, ok := oldObj.(*codey.CodeyInstance)
+	if !ok {
+		return nil, fmt.Errorf("not a valid CodeyInstance object")
+	}
+	newCodeyInstance, ok := newObj.(*codey.CodeyInstance)
+	if !ok {
+		return nil, fmt.Errorf("not a valid CodeyInstance object")
+	}
+
+	codeyFqdn := newCodeyInstance.ObjectMeta.Name + codeyUrlSuffix
+	if err := isCodeyFqdnUnique(codeyFqdn); err != nil {
+		return nil, fmt.Errorf("failed FQDN validation: %v", err)
+	}
+
+	return p.DefaultWebhookHandler.ValidateUpdate(ctx, oldObj, newObj)
+}
+
+// Checks if a given FQDN is already in use by some CodeyInstance in the cluster
+func isCodeyFqdnUnique(fqdn string) error {
+	config, err := rest.InClusterConfig()
```

Comment:
`DefaultWebhookHandler`, which is embedded into `CodeyInstanceWebhookHandler` already has a `client.Client`, which will have the right configurations. Please use that one instead of ad-hoc creating one. It will probably generate issues down the road with split control-plane and service-clusters.